sympy/ntheory/generate.py

```diff
@@ -244,11 +247,35 @@ def primepi(n):
         prime : Return the nth prime
     """
     n = int(n)
-    if n < 2:
-        return 0
-    else:
-        return sieve.search(n)[0]
-
+    if n<2:
+		return 0	
+	lim = int(n**0.5) #lim = floor(sqrt(n))
```

Comment:
Yes, it doesn't.
 Also, i have used a while loop next to `lim=  int(n**0.5)` to make it more accurate, even though I think it is not required for numbers that can be evaluated by the algorithm.